src/solc_0.8/polygon/child/land/PolygonLandTunnelMigration.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.2;
+
+import "../../../common/interfaces/IPolygonLandWithSetApproval.sol";
+import "../../../common/interfaces/IPolygonLandTunnel.sol";
+import "../../../common/interfaces/IERC721MandatoryTokenReceiver.sol";
+
+/// @title Tunnel migration on L2
+contract PolygonLandTunnelMigration is IERC721MandatoryTokenReceiver {
+    uint256 private constant GRID_SIZE = 408;
+
+    struct OwnerWithLandIds {
+        address owner;
+        uint256[] sizes;
+        uint256[] x;
+        uint256[] y;
+    }
+
+    IPolygonLandWithSetApproval public polygonLand;
+    address public newLandTunnel;
+    address public oldLandTunnel;
+    address private admin;
+
+    event TunnelLandsMigrated(address indexed oldLandTunnel, address indexed newLandTunnel, uint256[] ids);
+    event TunnelLandsMigratedWithWithdraw(OwnerWithLandIds _ownerWithLandIds);
+    event TunnelQuadsMigrated(
+        address indexed oldLandTunnel,
+        address indexed newLandTunnel,
+        uint256[] sizes,
+        uint256[] x,
+        uint256[] y
+    );
+    event AdminChanged(address _newAdmin);
+
+    modifier isAdmin() {
+        require(admin == msg.sender, "!AUTHORISED");
+        _;
+    }
+
+    /// @notice changes admin to new admin
+    /// @param _newAdmin the new admin to be set
+    function changeAdmin(address _newAdmin) external isAdmin {
+        admin = _newAdmin;
+        emit AdminChanged(_newAdmin);
+    }
+
+    constructor(
+        address _polygonLand,
+        address _newLandTunnel,
+        address _oldLandTunnel,
+        address _admin
+    ) {
+        admin = _admin;
+        polygonLand = IPolygonLandWithSetApproval(_polygonLand);
+        newLandTunnel = _newLandTunnel;
+        oldLandTunnel = _oldLandTunnel;
+    }
+
+    /// @dev Transfers all the passed land ids from the old land tunnel to the new land tunnel
+    /// @notice This method needs super operator role to execute
```

Comment:
`isAdmin` is not the same as super operator role ?

Reply:
No we have a single admin and we shouldn't make the migration contracts admin

Reply:
admin here is the admin of migration contract so that migration functions can only be called by the admin not any one else

Reply:
`isAdmin` is not really necessary i guess